### PR TITLE
Added Guzzle HTTP client

### PR DESCRIPTION
**Pull request by gmponos**
Added an HTTP Client for Guzzle

**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Comment by rvanlaak** [comment]
👏 for getting this merged

**Comment by gmponos** [comment]
Ping

File: Twilio/Http/GuzzleClient.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        try {
+            $response = $this->client->send(new Request($method, $url, $headers), [
+                'timeout' => $timeout,
+                'auth' => [$user, $password],
+                'query' => $params,
+                'form_params' => $data,
+            ]);
+        } catch (BadResponseException $exception) {
+            $response = $exception->getResponse();
+            return new Response($response->getStatusCode(), $response->getBody()->getContents(), $response->getHeaders());
```
Comment (childish-sambino) [inline Twilio/Http/GuzzleClient.php]: This line isn't needed.

File: Twilio/Tests/Unit/Http/GuzzleClientTest.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    public function setUp()
+    {
+        parent::setUp();
+        $this->mockHandler = new MockHandler();
+        $this->client = new GuzzleClient(new Client([
+            'handler' => HandlerStack::create($this->mockHandler),
+        ]));
+    }
+
+    public function testPostMethod()
```
Comment (childish-sambino) [inline Twilio/Tests/Unit/Http/GuzzleClientTest.php]: A _4XX_ or _5XX_ test would help validate/document the `BadResponseException` logic.

File: Twilio/Http/GuzzleClient.php (hunk 1 above)
Comment (gmponos) [reply Twilio/Http/GuzzleClient.php]: Done.. :)

File: Twilio/Tests/Unit/Http/GuzzleClientTest.php (hunk 2 above)
Comment (gmponos) [reply Twilio/Tests/Unit/Http/GuzzleClientTest.php]: done :)

**Pull request merged**